src/cmaq_preprocess/bias.py

```diff
@@ -44,6 +51,27 @@ def earliest_mean(
             date += one_day
     raise ValueError("no valid observations found")
 
+def earliest_region(
+    start_date: datetime.date,
+    end_date: datetime.date,
+    obs_file: pathlib.Path,
+) -> tuple:
+    obs = ObservationData.from_file(obs_file)
+    # now find the earliest date with obs and return their mean
+    one_day = datetime.timedelta(days=1)
+    date = start_date
+    while date <= end_date:
+        date_string = date.strftime("%Y%m%d")
+        if len(obs.ind_by_date[date_string]) > 0:
```

Comment:
This part of `earliest_region` is so identical to `earliest_mean`, I wonder if instead we need a helper function like `earliest_observation_date` which returns a `Date` that can be used in the parent function to extract the mean and the extent.